src/ngMock/angular-mocks.js

```diff
@@ -1511,15 +1511,25 @@ function createHttpBackendMock($rootScope, $timeout, $delegate, $browser) {
     }
 
     if (expectation && expectation.match(method, url)) {
+      var expectationError;
+
       if (!expectation.matchData(data)) {
-        throw new Error('Expected ' + expectation + ' with different data\n' +
-            'EXPECTED: ' + prettyPrint(expectation.data) + '\nGOT:      ' + data);
+        expectationError = new Error('Expected ' + expectation + ' with different data\n' +
+          'EXPECTED: ' + prettyPrint(expectation.data) + '\n' +
+          'GOT:      ' + data);
       }
 
       if (!expectation.matchHeaders(headers)) {
```

Comment:
Since we are now not throwing the `if` block above, this should be an `else if`.
(Is there a test that catches this?)

Reply:
Now, there is. :) It breaks nothing. If both data and headers don't match, we still get an error. Do you insist on `else if`?